pgmpy/datasets/datasets.py

```diff
@@ -0,0 +1,342 @@
+
+
+def load_dataset(
+    name: str, load_ground_truth: bool = True
+) -> Tuple[pd.DataFrame, Optional[DAG]]:
+    ds = DATASETS.get(name)
+    if ds is None:
+        raise ValueError(
+            f"Dataset '{name}' not found. Available datasets: {DATASETS.list_all()}"
+        )
+
+    try:
+        X = ds.load()
+    except TypeError:
+        X = ds.load()
+
+    gt = ds.load_ground_truth() if load_ground_truth else None
+
+    if gt is None:
+        gt = DAG()
+
+    data_cols = set(X.columns)
+    gt_nodes = set(gt.nodes())
+
+    if data_cols != gt_nodes:
+
+        missing_in_gt = data_cols - gt_nodes
+        gt.add_nodes_from(missing_in_gt)
+
+        missing_in_data = gt_nodes - data_cols
+        if missing_in_data:
+            import warnings
+
+            warnings.warn(
+                f"Ground truth for '{name}' has nodes not in data: {missing_in_data}"
+            )
+
+    return X, gt
+
+
+@dataset_class
+class Abalone(BaseDataset):
+    name = "abalone"
+    tags = {
+        "has_ground_truth": True,
+        "is_simulated": False,
+        "n_variables": 9,
+        "n_samples": 4177,
+        "is_discrete": False,
+        "is_continuous": False,
+        "is_mixed": True,
+        "is_ordinal": False,
+    }
+
+    base_url = "https://raw.githubusercontent.com/cmu-phil/example-causal-datasets/main/real/abalone/"
+
+    VARIANT_URLS = {
+        "continuous": base_url + "data/abalone.continuous.txt",
+        "mixed_numeric": base_url + "data/abalone.mixed.numeric.txt",
+        "mixed_max3": base_url + "data/abalone.mixed.maximum.3.txt",
+    }
+    GROUND_TRUTH_URL = base_url + "ground.truth/abalone.knowledge.txt"
+    DEFAULT_VARIANT = "mixed_numeric"
+
+    @classmethod
+    def _fetch(cls, key: str) -> bytes:
+        if key.startswith("data:"):
+            variant = key.split(":", 1)[1]
+            url = cls.VARIANT_URLS.get(variant)
+            if url is None:
+                raise KeyError(
+                    f"Unknown abalone variant '{variant}'. Available: {list(cls.VARIANT_URLS.keys())}"
+                )
+        elif key == "ground_truth":
+            url = cls.GROUND_TRUTH_URL
+        else:
+            raise KeyError(f"Unknown fetch key '{key}'.")
+        resp = requests.get(url, timeout=60)
+        resp.raise_for_status()
+        return resp.content
+
+    @classmethod
+    def load(cls, variant: Optional[str] = None) -> pd.DataFrame:
+        v = variant or cls.DEFAULT_VARIANT
+        raw = cls.load_or_fetch(f"data:{v}")
+        return load_df_from_txt(raw)
+
+    @classmethod
+    def load_ground_truth(cls) -> Optional[DAG]:
+        raw = cls.load_or_fetch("ground_truth")
+        return parse_knowledge_txt(raw)
+
+
+def load_abalone(
+    variant: Optional[str] = None, load_ground_truth: bool = True
+) -> Tuple[pd.DataFrame, Optional[DAG]]:
+    df = Abalone.load(variant=variant)
+    gt = Abalone.load_ground_truth() if load_ground_truth else None
+    return df, gt
```

Comment:
I don't think we should implement these individual `load_*` methods. The `load_dataset` method should handle all loading by name.